### PR TITLE
Fix handling of execute reply and add execute request options

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: src/ikernel.ts [hunk 1]
```diff
@@ -1047,34 +1047,112 @@ namespace KernelMessage {
...
+
+    /**
+     * Whether to execute the code as quietly as possible.
+     * The default is `false`.
+     */
     silent?: boolean;
+
+    /**
+     * Whether to store history of the execution.
+     * The default `true` is silent is False.
```
Comment (minrk) [inline src/ikernel.ts]: s/is silent/if silent/

File: src/ikernel.ts [hunk 2]
```diff
@@ -1047,34 +1047,112 @@ namespace KernelMessage {
...
+     * It is forced to  `false `if silent is `true`.
+     */
     store_history?: boolean;
+
+    /**
+     * A mapping of names to expressions to be evaluated in the
+     * user's dict.
```
Comment (minrk) [inline src/ikernel.ts]: s/user's dict/kernel's interactive namespace/

File: src/ikernel.ts (hunk 1 above)
Comment (blink1073) [inline src/ikernel.ts]: Fixed

File: src/ikernel.ts (hunk 2 above)
Comment (blink1073) [inline src/ikernel.ts]: Fixed

**Comment by minrk** [comment]
Thanks!

**Pull request merged**